### DISCUSSION TITLE
Trouble testing with fork

### DISCUSSION THREAD

**Discussion by forgot** [feathersjs/feathers#2346] [Q&A]
I've been working with the maintainer of Grant to account for Apple's non-standard OAuth, and I'm trying to test it out within Feathers. Unfortunately I'm fairly new to NPM, and I'm running into issues trying to get things to build. 

They've asked that I test simov/grant@15d64bd913474764a47ee4a0e5fce10dc55e1b6c, but I'm unsure how to pull this into my [fork](https://github.com/forgot/feathers) of Feathers to test. The only change is pointing the Grant dependency in `authentication-oauth` to [use that specific commit](https://github.com/forgot/feathers/commit/28cdb0d6aa5eb08a291bacf610d5e0e37cf8c536).

My problem is that when I try and use my fork of Feathers, `npm install` fails. The dependencies section of my `package.json` looks like this:
```
  "dependencies": {
    "@feathersjs/authentication": "^5.0.0-pre.3",
    "@feathersjs/authentication-local": "^5.0.0-pre.3",
    "@feathersjs/authentication-oauth": "forgot/feathers.git#28cdb0d6",
    "@feathersjs/configuration": "^5.0.0-pre.3",
    "@feathersjs/errors": "^5.0.0-pre.3",
    "@feathersjs/express": "^5.0.0-pre.3",
    "@feathersjs/feathers": "^5.0.0-pre.3",
    "@feathersjs/socketio": "^5.0.0-pre.3",
    "@feathersjs/transport-commons": "^5.0.0-pre.3",
    "@types/config": "0.0.36",
    "compression": "^1.7.4",
    "cors": "^2.8.5",
    "dotenv": "^8.2.0",
    "feathers-hooks-common": "forgot/feathers-hooks-common.git#Dove",
    "feathers-sequelize": "forgot/feathers-sequelize.git#Dove",
    "helmet": "^4.1.1",
    "pg": "^8.4.1",
    "sequelize": "^6.3.5",
    "serve-favicon": "^2.5.0",
    "winston": "^3.3.3"
  }
```

When I run `npm install --verbose` it fails with the following:
```
npm ERR! code 127
npm ERR! path /path/to/project/node_modules/@feathersjs/authentication-oauth
npm ERR! command failed
npm ERR! command sh -c lerna bootstrap
npm ERR! sh: lerna: command not found
npm verb exit 127
```

I tried adding `"lerna": "^4.0.0"` to my `devDependencies`, and ran `npm install --verbose` again, and it still fails with the following:
```
npm ERR! lerna info lifecycle @feathersjs/feathers@5.0.0-pre.3~prepublish: Failed to exec prepublish script
npm ERR! lerna ERR! lifecycle "prepublish" errored in "@feathersjs/feathers", exiting 1
npm verb exit 1
npm timing npm Completed in 707262ms
npm verb code 1
``` 

Again, I'm new to this, and couldn't find any answers on the web, so I thought I'd ask here. Any ideas on how to get this to work?

**Comment by daffl** [comment]
Unfortunately due to the monorepo and TypeScript setup it is not easily possible to directly link to a fork on GitHub. You can test locally by cloning the repository, installing the dependencies:

```
git clone git@github.com:forgot/feathers.git
cd feathers
git checkout 28cdb0d6
npm install
```

And then linking the package:

```
cd packages/authentication-oauth
npm link
```

Then in your test repository run:

```
npm link @feathersjs/authentication-oauth
```

If you make changes make sure to run `npm run compile` in `packages/authentication-oauth`.
